llama_index/llms/ollama.py

```diff
@@ -70,38 +67,132 @@ def _model_kwargs(self) -> Dict[str, Any]:
             **self.additional_kwargs,
         }
 
-    def _get_all_kwargs(self, **kwargs: Any) -> Dict[str, Any]:
-        return {
-            **self._model_kwargs,
-            **kwargs,
-        }
-
-    def _get_input_dict(self, prompt: str, **kwargs: Any) -> Dict[str, Any]:
-        return {self.prompt_key: prompt, **self._model_kwargs, **kwargs}
-
     @llm_chat_callback()
     def chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
```

Comment:
Not for this PR probably, but if we are refactoring things here, can we use httpx to implement async endpoints as well?

I'm sure there's some common code for creating requests that could be shared between them as well 🤔

Reply:
Yeah I feel the code is a bit repetitive, and using httpx is probably better since it supports async. Do you feel comfortable if we keep using requests in this PR and have a follow up PR to change to httpx? Should we use httpx for all clients that are using requests or just for Ollama?

Reply:
I usually just use httpx for async requests lol. I think it's fine to mix.